File: app/src/main/java/me/kyuubiran/qqcleaner/activity/AppCompatTransferActivity.java
package me.kyuubiran.qqcleaner.activity;

import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

//From QNotified
public class AppCompatTransferActivity extends AppCompatActivity {
    @Override
    public ClassLoader getClassLoader() {
        return AppCompatTransferActivity.class.getClassLoader();
    }

    @Override
    protected void onRestoreInstanceState(Bundle savedInstanceState) {
        if (savedInstanceState != null) {
            Bundle windowState = savedInstanceState.getBundle("android:viewHierarchyState");
            if (windowState != null) {
                windowState.setClassLoader(AppCompatTransferActivity.class.getClassLoader());
            }
        }
        super.onRestoreInstanceState(savedInstanceState);
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }
}

File: app/src/main/java/me/kyuubiran/qqcleaner/utils/resinjection/transferactivity/AppCompatTransferActivity.kt
package me.kyuubiran.qqcleaner.utils.resinjection.transferactivity

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

/**
 * 所有在宿主内启动的AppCompatActivity都应该继承与此类 否则会报错
 */
open class AppCompatTransferActivity : AppCompatActivity() {
    override fun getClassLoader(): ClassLoader {
        return AppCompatTransferActivity::class.java.classLoader!!
    }

    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
        val windowState = savedInstanceState.getBundle("android:viewHierarchyState")
        windowState?.let {
            it.classLoader = AppCompatTransferActivity::class.java.classLoader!!
        }
        super.onRestoreInstanceState(savedInstanceState)
    }
}